dev/maven-plugin/src/main/scala/com/lightbend/lagom/maven/ScalaClassLoaderManager.scala

```diff
@@ -0,0 +1,63 @@
+package com.lightbend.lagom.maven
+
+import java.net.URLClassLoader
+import javax.inject.{ Inject, Singleton }
+
+import org.eclipse.aether.artifact.Artifact
+
+/**
+ * Implements sharing of Scala classloaders, to save on memory
+ */
+@Singleton
+class ScalaClassLoaderManager @Inject() (logger: MavenLoggerProxy) {
+
+  /**
+   * The list of Scala libraries. None of these libraries may have a dependency outside of this list, otherwise there
+   * will be classloading issues.
+   *
+   * Note that while adding more libraries to this list will allow more to be shared, it may also mean that classloaders
+   * can be shared in less cases, since it becomes less likely that there will be an exact match between two projects
+   * in what can be shared.
+   */
+  private val ScalaLibs = Set(
+    "org.scala-lang" -> "scala-library",
+    "org.scala-lang" -> "scala-reflect",
+    "org.scala-lang.modules" -> "scala-xml",
+    "org.scala-lang.modules" -> "scala-parser-combinators",
+    "org.scala-lang.modules" -> "scala-java8-compat"
+  )
+
+  private val ScalaVersionPattern = "_\\d+\\.\\d+.*$".r
+  private def stripScalaVersion(artifactId: String) = ScalaVersionPattern.replaceFirstIn(artifactId, "")
+
+  private def createCacheKey(artifacts: Seq[Artifact]): String = {
+    artifacts.map { artifact =>
+      import artifact._
+      s"$getGroupId:$getArtifactId:$getVersion"
+    }.sorted.mkString(",")
+  }
+
+  private var cache = Map.empty[String, ClassLoader]
```

Comment:
Yes and no. There's no maven console, you don't start maven, run it in dev mode, stop it, run tests, run it in dev mode again, etc, all in the one process.  Unlike sbt, in maven you only ever do one thing, then the process terminates.  So classloader leaks are not nearly as much of a problem.